### PR TITLE
CupertinoPicker part 2 - create a ListWheelScrollController

### PR DESCRIPTION
**Pull request by xster**
#7374

Diffbases: #13783

Diff: https://github.com/xster/flutter/compare/list-wheel...xster:list-wheel-position

### CODE REVIEW

File: packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+  ///
+  /// Jumps the item index position from its current value to the given value,
+  /// without animation, and without checking if the new value is in range.
+  void jumpToItem(int itemIndex) {
+    final _FixedExtentScrollPosition position = this.position;
+    position.jumpTo(itemIndex * position.itemExtent);
+  }
+
+  @override
+  ScrollPosition createScrollPosition(ScrollPhysics physics, ScrollContext context, ScrollPosition oldPosition) {
```
Comment (xster) [inline packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart]: wasn't sure if you meant making this covariant in ScrollController

**Comment by Hixie** [comment]
GitHub makes commenting on this very hard so I'll wait until part 1 has landed before commenting on this (or if you want to just merge all 4 parts into one, I can just comment on that, which might be easier for both of us).

**Comment by xster** [comment]
Ya, I wish github deals with diffbases. Feel free to just comment on part 4 (minus stuff you already commented on part 1). I just split it into 4 so you don't have to look at a 2k+ line in one shot but it might be logistically easier.

**Comment by Hixie** [comment]
Whether the patch is split into 1, 4, or 16 parts makes no difference to the amount of work I must do to review it. :-) (It might even be less work if it's all together, unless the split is for really separate parts, but then it would just be separate PRs, not a chain of them.)

**Comment by xster** [comment]
Closing to consolidate reviews into #14091

**Pull request closed without merging**